deepr/layers/core.py

```diff
@@ -160,3 +160,17 @@ def forward(self, tensors, mode: str = None):
             use_bias=self.use_bias,
             **self._kwargs
         )
+
+
+class Softmax(base.Layer):
+    """Apply softmax to the last dimension of tensor with filtering masked values"""
+
+    def __init__(self, n_in: int = 2, n_out=1, **kwargs):
+        super().__init__(n_in=n_in, n_out=n_out, **kwargs)
```

Comment:
maybe add a test for that class (masking seems important to get right)

Reply:
yes, good point

Reply:
added tests